packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix.vue

```diff
@@ -125,6 +138,59 @@ const parametersValueMap = computed(() => {
 	return result;
 });
 
+// Copy and paste utilities
+let timerId = -1;
+const clipboardText = ref('');
+const updateByMatrixBulk = (matrixToUpdate: MiraMatrix, text: string) => {
+	const parseResult = dsvParse(text);
+
+	// FIXME: Not very efficient, maybe emit in bulk rather than one-by-one
+	matrixToUpdate.forEach((row) => {
+		row.forEach(async (matrixEntry) => {
+			// If we have label information, use them as they may be more accurate, otherwise use indices
+			if (parseResult.hasColLabels && parseResult.hasRowLabels) {
+				const match = parseResult.entries.find(
+					(entry) => entry.rowLabel === matrixEntry.rowCriteria && entry.colLabel === matrixEntry.colCriteria
+				);
+				if (match) {
+					emit('update-cell-value', {
+						variableName: matrixEntry.content.id,
+						newValue: match.value,
+						mathml: (await pythonInstance.parseExpression(`${match.value}`)).mathml
+					});
+				}
+			} else {
+				const match = parseResult.entries.find(
+					(entry) => entry.rowIdx === matrixEntry.row && entry.colIdx === matrixEntry.col
+				);
+				if (match) {
+					emit('update-cell-value', {
+						variableName: matrixEntry.content.id,
+						newValue: match.value,
+						mathml: (await pythonInstance.parseExpression(`${match.value}`)).mathml
+					});
+				}
+			}
+		});
+	});
+};
+
+const pasteItemProcessor = async (item: DataTransferItem) => {
+	if (item.kind !== 'string') return;
+	if (item.type !== 'text/plain') return;
+
+	console.log('paste paste');
+
+	// Presume this is a full matrix, with row/column labels
+	item.getAsString(async (text) => {
+		updateByMatrixBulk(matrix.value, text);
+	});
+};
+const processPasteEvent = pasteEventGenerator(pasteItemProcessor);
+const pasteBuffer = (text: string) => {
```

Comment:
I am not following why we `pasteBuffer` and `pasteItemProcessor`
One if for me hitting the `paste button` and another is for me hitting `control v`? Why can these not be the same handler?

Reply:
One is for clipboard-buffer, the other paste-event. I will rename pasteBuffer.